### ISSUE TITLE
Junit4 analyze all test by default without skippy runner

### ISSUE THREAD

**Issue by Link184** [skippy-io/skippy#189]
It is a bit annoying to add rules to all the tests we want to cover with skippy, it can be especially painful on big projects that contains thousands of test classes.

We can customize gradle test tasks to ignore tests by overriding gradle `Test` task:

```kts
tasks.withType<Test> {
    useJUnitPlatform()
    testsToBeIgnored.forEach(::exclude)
}
```

Is it achievable?

**Comment by fmck3516** [comment]
> Is it achievable?

Not 100% sure. But: The ask makes sense - I will look into it.

**Comment by fmck3516** [comment]
This seems like a good fit for AOP-based solution. Conversely speaking: I'm not convinced that this is something that Skippy should try to solve.

Example using AspectJ's [@DeclareMixin](https://eclipse.dev/aspectj/doc/released/adk15notebook/ataspectj-itds.html#atDeclareMixin):

`build.gradle`:
```
plugins {
    ...
    id "io.freefair.aspectj.post-compile-weaving" version "8.6"
}

dependencies {
    ...
    implementation 'org.aspectj:aspectjrt:1.9.9'
    runtimeOnly 'org.aspectj:aspectjweaver:1.9.9'
}
```

`src/test/java/SkippyAspect.java`:
```
package com.example;

import io.skippy.junit4.Skippy;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.DeclareMixin;
import org.junit.Rule;
import org.junit.rules.TestRule;

@Aspect
public class SkippyAspect {

    @DeclareMixin("(*..*Test)")
    public static PredictWithSkippy skippyMixin() {
        return () -> Skippy.predictWithSkippy();
    }

    public interface PredictWithSkippy {
        @Rule
        TestRule getTestRule();
    }

}
```

Disclaimer: The above is not a polished recommendation (since it does not restrict weaving to the test code).

Thoughts on this approach? I definitely want to touch on this in the documentation.

**Comment by Link184** [comment]
> Thoughts on this approach? I definitely want to touch on this in the documentation.

I would consider bytecode manipulations as a last resort solution. Because it is hard to debug, maintain and test. We also can't be 100% sure that it will work on all jvms. In case we decide to use this solution then I would say it is much safer to deliver it as a separate tool(a plugin for a plugin, lol), to not include it in skippy.

**Comment by fmck3516** [comment]
> > Thoughts on this approach? I definitely want to touch on this in the documentation.
> 
> I would consider bytecode manipulations as a last resort solution. Because it is hard to debug, maintain and test. We also can't be 100% sure that it will work on all jvms. In case we decide to use this solution then I would say it is much safer to deliver it as a separate tool(a plugin for a plugin, lol), to not include it in skippy.

Yeah, I am definitely not considering to include this in Skippy. That's not a hill I want to die on (and not the right problem for Skippy to solve).

**Comment by fmck3516** [comment]
Closed - I see no clear path forward to implement this using JUnit 4 at this point. Please re-open if you want to toss more ideas around.
